internal/xdscache/v3/listener.go

```diff
@@ -502,7 +502,11 @@ func (c *ListenerCache) OnChange(root *dag.DAG) {
 
 				filters = envoy_v3.Filters(cm)
 
-				alpnProtos = envoy_v3.ProtoNamesForVersions(cfg.DefaultHTTPVersions...)
+				if len(vh.HTTPVersions) != 0 {
```

Comment:
we need to make sure to have unit tests for this precedence behavior, i.e. when http versions are set in the virtualhost, they override the default